addons/google_calendar/models/google_sync.py

```diff
@@ -298,7 +298,12 @@ def _get_sync_partner(self, emails):
                      email not in [partner.email_normalized for partner in partners]]
         if remaining:
             partners += self.env['mail.thread']._mail_find_partner_from_emails(remaining, records=self, force_create=True, extra_domain=[('type', '!=', 'private')])
-        return partners
+        unsorted_partners = self.env['res.partner'].browse([p.id for p in partners])
+        result = []
+        # partners needs to be sorted according to the emails order provided by google
+        for email in emails:
+            result.append(unsorted_partners.filtered(lambda p: p.email_normalized == email))
```

Comment:
```suggestion
        unsorted_partners = self.env['res.partner'].browse([p.id for p in partners])
        # partners needs to be sorted according to the emails order provided by google
        k = {v: n for n, v in enumerate(emails)}
        results = unsorted_partners.sorted(key=lambda p: k.get(p.email_normalized, -1))
```
So we can avoid loop into loop